types/vue.d.ts

```diff
@@ -2,17 +2,18 @@
  * Augment the typings of Vue.js
  */
 
-import Vue, { ComponentOptions } from "vue";
-import { MetaInfo } from './index';
+import Vue, { ComponentOptions } from 'vue'
+import { MetaInfo, VueMetaPlugin } from './vue-meta'
 
-declare module "vue/types/options" {
-  interface ComponentOptions<V extends Vue> {
-    metaInfo?: MetaInfo | (() => MetaInfo)
+declare module 'vue/types/vue' {
+  interface Vue {
+    $meta(): VueMetaPlugin
   }
 }
 
-declare module "vue/types/vue" {
-  interface Vue {
-    $meta(): MetaInfo
+declare module 'vue/types/options' {
+  interface ComponentOptions<V extends Vue> {
+    metaInfo?: MetaInfo | (() => MetaInfo)
```

Comment:
@kevinmarrec The `metaInfo?` here wont be correct for eg Nuxt as Nuxt uses `head` as `VueMetaOptions.keyName`. Do you know of a way to make the `metaInfo` here dynamic and to point it to the keyName the VueMetaPlugin was instantiated with? Would something like `[VueMetaOptions.keyName]` work? I guess not at thats just an abstract definiton, maybe something like `[V.getOptions().keyName]` (although that doesnt work because then I get `TS1169: A computed property name in an interface must refer to an expression whose type is a literal type or a 'unique symbol' type.`?

Reply:
The above also applies to the `vmid?` I use in the just added `MetaDataProperty` intreface, its vmid by default but Nuxt uses `hid`

Reply:
@pimlie Sorry i'm late, hum are types good now ? I checked and it seems good, is there any issue still present you mentioned above ?

Reply:
@kevinmarrec np, I think there is still an issue but am not 100% sure. As said, the property name on the Vue component is configurable when you install the Vue-Meta plugin and eg for Nuxt `keyName: 'head'` (see docs here: https://vue-meta.nuxtjs.org/guide/#options). I think that means these types will now not work on Nuxt projects right? Because the types only use the default value for keyName and not the value `head` Nuxt uses?

Reply:
@pimlie alright I see, well I think there is no proper solution, that's the issue when your library provides feature like being able to customize names of the API ^^

Reply:
Yeah, we will probably remove that functionality in v3. Do you think it would be helpful to duplicate them and also add the property names Nuxt uses? 

Eg so we get:
```ts
export interface MetaDataProperty {
  vmid?: string,
  hid?: string,
  [key: string]: any
}
```
and
```ts
declare module 'vue/types/options' {
  interface ComponentOptions<V extends Vue> {
    metaInfo?: MetaInfo | MetaInfoComputed
    head?: MetaInfo | MetaInfoComputed
  }
}
```

or will that mess up other things?

Reply:
I think we should extend them through Nuxt ones, in Nuxt `packages/vue-app/types`. `head` is already defined here btw.

Something like
```ts
declare module 'vue-meta' {
  interface MetaDataProperty {
    hid?: string
    }
}
```
And do that way for each needed interface.

Reply:
Ah yes, that sounds logical. If you dont have any other recommendations I will proceed with merging this

Reply:
We're good to merge, nice work again :)

Reply:
Thanks for your help & insights, couldnt have done it without you!